src/react-apps/applications/ux-editor/src/containers/EditContainer.tsx

```diff
@@ -261,7 +261,14 @@ class Edit extends React.Component<IEditContainerProps, IEditContainerState> {
         inEditMode: false,
       },
     }, () => {
-      this.handleSaveChange(this.state.component);
+      console.log(this.state.component, this.props.component);
+      /* tslint:disable:no-var-keyword prefer-const */
+      var {required, ...restState} = this.state.component;
```

Comment:
Very open for suggestions

Reply:
Why use var instead of const?

Reply:
The challenge: We need to check if the component should be updated when clicking on the "check" button. And the variables we're interested in are usually dataModelBinding, title or description. When changing the dataModelBinding the "required" may change depending on the data model. To prevent "required" to be set back to what's set by the data model, when clicking on "check", we need to check if anything but the "required" field has changed.

Using var because its more open than const, which means I'm allowed to declare "required" more than once. And as mentioned above I need to check all variables but "required".

Reply:
I'm unsure what you mean. In the scope the var is set, it is never reassigned. So its no point declaring it a var. Thats why you get a tslint complaint. Maybe you can show me tomorrow.